scripts/skills/perks/perk_rf_the_rush_of_battle.nut

```diff
@@ -66,6 +66,14 @@ this.perk_rf_the_rush_of_battle <- ::inherit("scripts/skills/skill", {
 		}
 	}
 
+	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
+	{
+		if (_skill.isAttack())
```

Comment:
Are there any attack skills that can target the actor himself? I think we can and should assume that this cannot be the case because I believe the code of the vanilla game wherever they check for `skill.isAttack()` is also written with this assumption. The check in `onBeforeDamageReceived` is to account for things like bleeding/poison damage in which case the source of the damage is the actor receiving the damage.

Reply:
Nevermind I thought this function was called whenever any entity on the battlefield uses any skill, because I remember how other classes who use this function always check `_skill == this`.
This seems to be correct